### ISSUE TITLE
Terminal escape characters in output

### ISSUE THREAD

**Issue by kflak** [luisjure/csound-vim#8]
Hi,

I'm dipping my toes into the csound waters here... Something is happening to the output in vim, which doesn't happen when I run a csd file in the terminal, getting a lot of `^[[m` characters at the beginning of certain lines:

```
:!csound -d -o devaudio "/home/kf/csound/test.csd" 
UnifiedCSD:  /home/kf/csound/test.csd
Elapsed time at end of orchestra compile: real: 0.001s, CPU: 0.001s
^[[msorting score ...
	... done
Elapsed time at end of score sort: real: 0.001s, CPU: 0.001s
^[[m--Csound version 6.18 (double samples) Apr  7 2023
[commit: none]
^[[mlibsndfile-1.2.2
^[[mdisplays suppressed
sr = 48000.0,^[[m kr = 48000.000,^[[m ksmps = 1
^[[m0dBFS level = 1.0,^[[m A4 tuning = 440.0
^[[morch now loaded
^[[maudio buffered in 256 sample-frame blocks
^[[m
ALSA output: total buffer size: 1024, period size: 256
writing 256 sample blks of 64-bit floats to devaudio
SECTION 1:
^[[mnew alloc for instr 1:
^[[m
B  0.000 ..  2.000 T  2.000 TT  2.000 M:  0.11577
Score finished in csoundPerform().
inactive allocs returned to freespace
end of score.		   overall amps:^[[m  0.11577
	   overall samples out of range:^[[m        0^[[m
0 errors in performance
^[[mElapsed time at end of performance: real: 2.278s, CPU: 0.027s
^[[m
```

I suspect there's some underlying terminal issue here... Running this on NVIM v0.10.0-dev-741+g996dd36c77, using the `kitty` terminal emulator and Arch Linux, kernel version 6.5.3. 

EDIT: testing it in `alacritty` and `xfce4-terminal` yields the same result, so it's not specific to Kitty.

**Comment by luisjure** [comment]
Hi, this is an issue on how gvim manages terminal output, and isn't related to the plugin itself. Actually, it shouldn't happen on plain vim.

**Comment by kflak** [comment]
Nope! Not using gvim, using neovim ;-) Tried again with a completely minimal config, and I still get the same issue. Maybe there's a difference in how nvim and vim handle terminal output?

**Comment by luisjure** [comment]
Hi, sorry, I'm not familiar with Neovim and I don't know how it relates to Vim/gVim. In any case, I don't see that this issue is related to the plugin, but rather it's caused by how each of those programs handle terminal output. As I said, try regular Vim in a terminal, and you should see clean output.
I'm closing the issue now, because I can't see how I could do something about that from the plugin.

**Comment by kflak** [comment]
No worries, will take a look into it when I have a bit more time. Not even 100% sure I can allow myself to spend time in csound, as there's just too much other stuff that needs attention right now.

**Comment by kflak** [comment]
OK, couldn't help myself, so I checked it out a bit. Seems that in neovim the `:! someCommand` produces raw output with no interpretation of ansi escape characters. In order to get the interpreted version you need to use `:terminal someCommand` instead.  This creates a new terminal buffer with the correct output. If I get some time I'll see if I can implement something for this, maybe do a check if we're in neovim or in vim and set the macros accordingly...

**Comment by luisjure** [comment]
:term[inal] is a standard vim command. I could simply replace :! by :term, and the macros would work just as well. The behavior would be a bit different, though, and you need to type :q to leave the terminal, instead of simply pressing any key.
I may change the default macros later, but in the meantime you can personalize your macros by including the file mycsound_macros under macros/, see the [README](https://github.com/luisjure/csound-vim#user-defined-macros).

**Comment by kflak** [comment]
Ah, nice idea to use userdefined macros. I only see one problem with this, though, which is that this will cause the update process to go sideways whenever you push a new version of the plugin... Maybe it would be better to expose a set of functions and commands to the user and allow them to set their own keyboard shortcuts in the config? That way you'll get a better separation between things, keeping the plugin untouched, and all userdefined keyboard shortcuts where they belong... The default keyboard bindings could still be the same, but instead of pointing to macros they would call the appropriate function instead.

EDIT: Alternatively, maybe it would be a good idea to expose the `mymacros` file path as a `g:csound-vim-macro-path` or something like this, so that the macros can be picked up from anywhere in the file system instead of buried deep in the plugin? Just a thought...

**Comment by luisjure** [comment]
Hi, sorry for the late response. I closed your pull request. I incorporated your code for neovim into a new version of the file `csound_macros` that takes care (I hope) of regular Vim on a terminal, of GVim and of Neovim (I trust you on this one, I'm not familiar with neovim and I don't have it installed to test).

I don't understand your concerns about `mycsound_macros` when updating the plugin, the user-defined files are included in `.gitignore`, so there's no problem whatsoever when pulling a new version. 

Anyway, you propose some interesting ideas. Right now I don't have the time to look into this but later on I might return to this. Thank you for your contribution.
